e2e/global/builder/registry_test.go

```diff
@@ -0,0 +1,89 @@
+//go:build integration
+// +build integration
+
+// To enable compilation of this file in Goland, go to "Settings -> Go -> Vendoring & Build Tags -> Custom Tags" and add "integration"
+
+/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package builder
+
+import (
+	"os"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+
+	. "github.com/apache/camel-k/e2e/support"
+)
+
+func TestRunWithDockerHubRegistry(t *testing.T) {
+	user := os.Getenv("TEST_DOCKER_HUB_USERNAME")
+	pass := os.Getenv("TEST_DOCKER_HUB_PASSWORD")
+	if user == "" || pass == "" {
+		t.Skip("no docker hub credentials: skipping")
+		return
+	}
+
+	WithNewTestNamespace(t, func(ns string) {
+		operatorID := "camel-k-docker-hub"
+		Expect(KamelInstallWithID(operatorID, ns,
+			"--registry", "docker.io",
```

Comment:
Is `--registry` an operator switch or a platform switch?
If under global, an operator switch becomes problematic as the operator is not reinstalled so that switch may not be re-applied. If platform then all is good because a new platform is always created in the new namespace.

Reply:
It's a platform switch encoded into `IntegrationPlatform`, so it should be ok.

As a side note, now `kamel install` seems to be a bit overloaded with those flags for different concerns. There's no easy way to distinguish between operator swtiches and platform switches. Maybe we should split the command into two: operator installation and platform installation.

Reply:
fyi @squakez @christophd @astefanutti